pandas-stubs/core/dtypes/dtypes.pyi

```diff
@@ -1,125 +1,71 @@
 
 class PandasExtensionDtype(ExtensionDtype):
     subdtype = ...
-    str: _str | None = ...
+    str: str | None = ...
     num: int = ...
     shape: tuple[int, ...] = ...
     itemsize: int = ...
     base = ...
     isbuiltin: int = ...
     isnative: int = ...
-    def __hash__(self) -> int: ...
+
     @classmethod
     def reset_cache(cls) -> None: ...
 
-class CategoricalDtypeType(type): ...
-
 class CategoricalDtype(PandasExtensionDtype, ExtensionDtype):
-    name: _str = ...
-    type: type[CategoricalDtypeType] = ...
-    kind: _str = ...
-    str: _str = ...
-    base = ...
     def __init__(
-        self, categories: Sequence[Any] | None = ..., ordered: Ordered = ...
+        self,
+        categories: Series | Index | list[Any] | None = ...,
+        ordered: Ordered = ...,
     ) -> None: ...
-    @classmethod
-    def construct_from_string(cls, string: _str) -> CategoricalDtype: ...
-    def __hash__(self) -> int: ...
-    def __eq__(self, other) -> bool: ...
-    @classmethod
-    def construct_array_type(cls): ...
-    @staticmethod
-    def validate_ordered(ordered: Ordered) -> None: ...
-    @staticmethod
-    def validate_categories(categories, fastpath: bool = ...): ...
-    def update_dtype(self, dtype: _str | CategoricalDtype) -> CategoricalDtype: ...
     @property
     def categories(self) -> Index: ...
     @property
     def ordered(self) -> Ordered: ...
 
 class DatetimeTZDtype(PandasExtensionDtype):
-    type: type[Timestamp] = ...
-    kind: _str = ...
-    str: _str = ...
-    num: int = ...
-    base = ...
-    na_value = ...
-    def __init__(self, unit: _str = ..., tz=...) -> None: ...
+    def __init__(
+        self, unit: Literal["ns"] = ..., tz: str | int | dt.tzinfo | None = ...
+    ) -> None: ...
     @property
-    def unit(self): ...
+    def unit(self) -> Literal["ns"]: ...
     @property
-    def tz(self): ...
-    @classmethod
-    def construct_array_type(cls): ...
-    @classmethod
-    def construct_from_string(cls, string: _str): ...
+    def tz(self) -> dt.tzinfo: ...
     @property
-    def name(self) -> _str: ...
-    def __hash__(self) -> int: ...
-    def __eq__(self, other) -> bool: ...
+    def na_value(self) -> NaTType: ...
 
 class PeriodDtype(PandasExtensionDtype):
-    type: type[Period] = ...
-    kind: _str = ...
-    str: _str = ...
-    base = ...
-    num: int = ...
-    def __new__(cls, freq=...): ...
-    @property
-    def freq(self): ...
-    @classmethod
-    def construct_from_string(cls, string: _str): ...
+    def __init__(self, freq: str | BaseOffset = ...): ...
     @property
-    def name(self) -> _str: ...
+    def freq(self) -> BaseOffset: ...
     @property
-    def na_value(self): ...
-    def __hash__(self) -> int: ...
-    def __eq__(self, other) -> bool: ...
-    @classmethod
-    def is_dtype(cls, dtype) -> bool: ...
-    @classmethod
-    def construct_array_type(cls): ...
+    def na_value(self) -> NaTType: ...
     def __from_arrow__(self, array): ...
```

Comment:
probably should delete `__from_arrow__()` as a user would never call it

Reply:
Done.